returnn/torch/data/pipeline.py

```diff
@@ -455,6 +466,74 @@ def __getitem__(self, index):
         raise Exception(f"{self.__class__.__name__}.__getitem__ not supported")
 
 
+class ShufflingDataPipe(torch.utils.data.IterDataPipe):
+    """Data pipe that shuffles batches while keeping certain data keys in order."""
```

Comment:
Extend the doc to explain how this differs from (or extends over) `torch.utils.data.datapipes.iter.Shuffler`.